src/workloads/docs/CrudActorFSM.yml

```diff
@@ -0,0 +1,148 @@
+        - WriteCommand: insertOne
+          Document: {phoneID: 2, state: "Off"}
+        - WriteCommand: insertOne
+          Document: {phoneID: 3, state: "Off"}
+        - WriteCommand: insertOne
+          Document: {phoneID: 4, state: "Off"}
+    - &nop {Nop: true}
+    - *nop
+
+- Name: StateSmartPhoneDemo
+  Type: Crud
+  Threads: 5
+  Database: StateTest
+  Phases:
+  - *nop
+  - Duration: 30 seconds
+    Collection: CrudDemo
+    States:
+    - Name: "On"  # The device is on
+
+      # The states support the keywords Repeat, SleepBefore, SleepAfter, (local) Rate, and
+      # Duration. These keywords behave the same as in Phase Control, but operate per state (i.e.,
+      # per actor instance as it goes through the state machine.
+      Repeat: 1
+      Operations:
+      # A state can have (optional) a set of operations that execute upon entering the state. This
+      # state updates a specific document to indicate it is in state On.
+
+      # Note: Following the analogy to phases -- phases may be blocking or non-blocking. States are
+      # always non-blocking. When the phase ends, the state machine will finish any open operations,
+      # and then exit. It if is waiting for a delay, it will exit immediately. This is required as
+      # there may be states with long delays.
+      # Question: Should we support having a single fire operation with repeating operations? Here I
+      # could invent a case where the following operation is repeat 1, but there is another
+      # operation that repeats once a second while in this state. I don't know how to represent
+      # that.  
+      - OperationName: updateOne
+        OperationCommand:
+          Filter: &filter {phoneID: {^ActorID: {}}}
+          Update:
+            $set: {state: "On"}
+
+      # Transitions specifies next states, delays, and operations. The On state can transition to
+      # each of the three other states
+      Transitions:
+      - To: "Off"  # Transition to this state
+
+        # Transitions are probabilistically chosen based on weight. A transition with weight 2 is
+        # twice as likely to be picked as a transition with weight 1. Weights must be greater than
+        # 0.
+        Weight: 1
+
+        # Transitions also support the phase keywords of sleepBefore, sleepAfter, and repeat.
+        # ^TimeSpec is a proposed generator to randomly generate acceptable TimeSpecs with units. It
+        # accepts any valid int or double generator or constant for value.
+        SleepBefore: {^TimeSpec: {value: {^RandomDouble: {min: 0, max: 15}}, unit: "seconds"}}
+
+    - Name: "Off"  # The device is Off
+      Operations:
+      - OperationName: updateOne
+        OperationCommand:
+          Filter: *filter
+          Update:
+            $set: {state: "Off"}
+      Transitions:  # Can only go to On from Off
+      - To: "On"
+        Weight: 1
+        SleepBefore: {^TimeSpec: {value: {^RandomDouble: {min: 0, max: 15}}, unit: "seconds"}}
```

Comment:
Should these generators be using RandomInt instead? We have a TimeSpec constructor from `int64_t` which might be easier to repurpose.

Reply:
I'm open to pushing here, but since this includes units, shouldn't it be a float? I feel like 3.43 seconds should be a valid timespec.

Reply:
Consider only allowing integer values but allowing for more precision with the units `{^TimeSpec: {seconds: {^Int}}`. I have little strong opinion here fwiw beyond trying at least a bit to avoid decimals if we can.